### ISSUE TITLE
# ISSUE: Sentence-Level Transcription Review

### ISSUE THREAD

**Issue by mahesharma1984** [mahesharma1984/diagnostic-automated-workflow#21]
**Priority:** High  
**Component:** Transcriber (`core.py`, `transcribe.py`)  
**Type:** Enhancement  

---

## Problem

The current transcription review system operates at the **word level** — it flags individual unclear words with `[UNCLEAR]` markers. However, this misses a critical failure mode: **the AI confidently transcribes entire phrases incorrectly**.

### Example: Desmond Week 6

**AI Transcribed:**
```
Jonas slowly goes from needing to be the same as everybody else 
to wanting to get someone out to be part of the long body.
```

**Actual (from image):**
```
Jonas slowly goes from wanting to be the same as everybody else 
to wanting to end Sameness and to be part of a loving family.
```

**Errors not flagged:**
- "needing" → "wanting"
- "get someone out" → "end Sameness"
- "long body" → "loving family"

None of these triggered `[UNCLEAR]` because the AI was visually confident. But the sentence doesn't make semantic sense ("part of the long body"?).

---

## Root Cause

Word-level confidence ≠ sentence-level accuracy.

The AI can be 95% confident about each individual word while producing a sentence that is semantically nonsensical. Current system has no mechanism to catch this.

---

## Proposed Solution

Replace word-level review with **sentence-level review**:

1. AI transcribes full text (best guess for everything)
2. AI returns a separate list of "sentences needing review" with confidence scores
3. Review loop shows full sentences, user can accept or rewrite entire sentence

### New Data Structure

```python
@dataclass
class SentenceReview:
    """A sentence flagged for human review"""
    sentence_text: str          # Current transcription
    line_start: int             # Start line in document
    line_end: int               # End line in document  
    confidence: float           # 0.0-1.0
    reason: str                 # Why flagged
```

### New Review Flow

```
[1/5] Lines 1-3 (confidence: 72%)
  "Jonas slowly goes from needing to be the same as everybody else 
   to wanting to get someone out to be part of the long body."

  Reason: unusual phrase detected ("long body")
  
  → Accept (Enter) or rewrite: _______
```

User types corrected sentence, entire sentence is replaced.

---

## Acceptance Criteria

- [ ] `SentenceReview` dataclass added to `core.py`
- [ ] Transcription prompt updated to return sentence-level confidence
- [ ] CLI review loop (`transcribe.py`) shows full sentences
- [ ] User can rewrite entire sentence with single input
- [ ] Backward compatible: JSON output still has clean `transcription` field
- [ ] Test on Desmond Week 6 — catches "long body" error

---

## Files to Modify

| File | Changes |
|------|---------|
| `core.py` | Add `SentenceReview` dataclass, modify `TranscriptionResult`, update prompt |
| `transcribe.py` | Replace word-level review loop with sentence-level |

---

## Out of Scope

- Hybrid word+sentence mode (keep it simple)
- Automatic semantic validation (future enhancement)
- Multi-page sentence spanning (sentences don't cross pages)

---

## Related

- `TRANSCRIPTION_FAILURE_ANALYSIS_AND_FIX.md` — original hallucination analysis
- `ISSUE_Transcription_Accuracy_Fix.md` — word-level improvements (predecessor)
